In [1]:
from scipy.stats import norm, t
import numpy as np

<hr style="color: #009933; border: solid 1px">
<span style="color: #009933;">unknown std, $s$</span>

# <span style="color: #2455C3;">Cell phone law</span>

Researches are interested in the effects of a law that penalizes talking on phone while driving. The researches randomly sampled 10 counties in a US state before and after the law was passed. They measured the total number of accidents due to cell phone use the year before and the year after the law. They hypothized that the law would reduce the number of accidents due the cell phone use.

## <span style="color: #85100F">Solution</span>

#### <span style="color: #85100F">1. Statating the hypothesis</span>

$$\begin{array}{cl}
H_0: & \mu_{\text{post}} \geq \mu_{\text{pre}}\\
 & \text{the law has no effect in the number of accidents duw to cell phone use}\\
H_A: & \mu_{\text{post}} < \mu_{\text{pre}}\\
 & \text{the law helps to reduce the number of accidents due to cell phone use}
\end{array}$$

#### <span style="color: #85100F;">2. Analyzing sample data</span>

In [17]:
# known data
# one-sampple left tailed test
n = 10.0
mu = 0
alpha = 0.05

In [5]:
# samples - number of car accidents in thousands due to cell phone use
pretest = np.array([8,7,6,9,10,5,7,11,8,7])
posttest = np.array([5,6,4,6,5,3,2,9,4,4])
diff = posttest - pretest 
diff

array([-3, -1, -2, -3, -5, -2, -5, -2, -4, -3])

In [16]:
# mean and s
xbar_D = diff.mean()
s_D = diff.std(ddof=1)
SE = s_D / np.sqrt(n)
print "xbar_D={:0.2f}, s_D={:0.2f}, SE={:0.2f}".format(xbar_D,
                                                        s_D, SE)

xbar_D=-3.00, s_D=1.33, SE=0.42


#### <span style="color: #85100F;">3. Test statistic calculation</span>

In [18]:
t_score = (xbar_D - mu) / SE
t_score

-7.1151247353788545

#### <span style="color: #85100F;">4. Critical point determination</span>

In [10]:
t_critical = t.ppf(alpha, n-1)
t_critical

-1.8331129326536337

#### <span style="color: #85100F;">5. Results interpretation</span>

<span style="color: #009933;">The <b>null hypothesis is rejected</b>, which means that <b>the law has effectively reduce the number of accidents due the use of cell phone</b></span>

1) <u>Descriptive statistics:</u>
    $$\begin{array}{ccc}
    \mu & = & 0\\
    \bar{x}_D & = & -3.0\\
    s_D & = & 1.33
    \end{array}$$

2) <u>Inferential statistics:</u>

t(9)=-7.2, p=.00003, one-tailed<br>
Confidence interval on the mean difference on number of accidents<br>
95% CI = (-3.95 to -2.05)

3) <u>Effect size measures:</u>

* Cohen's d = -2.25
* R^2 = .85

So the <b>85%</b> of the <b>difference in accidents due to cell phone use</b> for the sample of 10 counties are <b>due to</b> the <b>cell-phone law</b>.

In [20]:
# P-value
P_value = t.cdf(t_score, n-1)
print "P-value = {:0.5f}".format(P_value)

P-value = 0.00003


In [22]:
# Cohen's d
d = (xbar_D - mu) / s_D
print "d = {:0.2f}".format(d)

d = -2.25


In [23]:
# R^2
r_squared = t_score**2 / (t_score**2 + (n-1))
print "R^2 = {:0.2f}".format(r_squared)

R^2 = 0.85


In [26]:
# CI
t_char = t.ppf(1-alpha/2, n-1)
me = t_char * SE
print "95% CI = ({:0.2f} to {:0.2f})".format(xbar_D-me, xbar_D+me)

95% CI = (-3.95 to -2.05)
